src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/ClusterLocalRegistrar.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.GrainDirectory;
+
+namespace Orleans.Runtime.GrainDirectory
+{
+    /// <summary>
+    /// The registrar for the Cluster-Local Registration Strategy.
+    /// </summary>
+    internal class ClusterLocalRegistrar : IGrainRegistrar
```

Comment:
Ok, I will try to come up with something. For example, we could have both an async and a sync interface and use whichever one is right for that registrar.